contributing/guide-api-promotion.md

```diff
@@ -0,0 +1,97 @@
+# Promoting an API
+
+This guide describes the process for promoting an API in Crossplane from one
+version to the next. This process should be handled with care over multiple
+releases, as it has proven fairly easy for Crossplane's API promotions of the
+past to be problematic for both upgrades and downgrades, as demonstrated by
+[#6148], [#5932], and [#4400].
+
+An explanation of Crossplane's feature lifecycle can be found in the [Feature
+Lifecycle] docs page.
+
+## Core Issue
+
+The root cause we have encountered in past problematic API promotions is that we
+try to drop a particular version from a CRD while there may still be resources
+of that version stored in etcd. Kubernetes does not allow this action because it
+can result in possible data loss.
```

Comment:
It allows it. But it's protected by `status.storedVersions`. It needs somebody to remove a version from that slice AFTER cleaning up etcd from old instances.